extensions/smallrye-metrics/deployment/src/main/java/io/quarkus/smallrye/metrics/deployment/SmallRyeMetricsProcessor.java

```diff
@@ -286,10 +285,12 @@ public void transform(TransformationContext context) {
      */
     @BuildStep
     void reflectiveMethodsWithGauges(BeanArchiveIndexBuildItem beanArchiveIndex,
-            BuildProducer<ReflectiveMethodBuildItem> reflectiveMethods) {
+            BuildProducer<ReflectiveClassBuildItem> reflectiveClass) {
         for (AnnotationInstance annotation : beanArchiveIndex.getIndex().getAnnotations(GAUGE)) {
             if (annotation.target().kind().equals(AnnotationTarget.Kind.METHOD)) {
-                reflectiveMethods.produce(new ReflectiveMethodBuildItem(annotation.target().asMethod()));
+                reflectiveClass.produce(
+                        ReflectiveClassBuildItem.builder(annotation.target().asMethod().declaringClass().name().toString())
+                                .methods(true).constructors(true).build());
```

Comment:
It won't be problematic to register the classes several times? I think we usually build a `Set` when doing that.

Reply:
It will not be problematic because of how we [aggregate](https://github.com/quarkusio/quarkus/blob/e145e31077afbc4cb3fbc4defb8636e344ab5854/core/deployment/src/main/java/io/quarkus/deployment/steps/NativeImageReflectConfigStep.java#L146) the info.

But I'll indulge and add a `Set`